File: go/leetcode/editor/cn/[724]寻找数组的中心下标.go
package leetcode

// leetcode submit region begin(Prohibit modification and deletion)
func pivotIndex(nums []int) int {
	var preSums = make([]int, len(nums)+1)
	preSums[0] = 0
	for i := 1; i < len(preSums); i++ {
		preSums[i] = preSums[i-1] + nums[i-1]
	}

	for j := 0; j < len(nums); j++ {
		if preSums[j] == preSums[len(nums)]-preSums[j+1] {
			return j
		}
	}

	return -1

}

//leetcode submit region end(Prohibit modification and deletion)

File: scala/leetcode_scala/src/main/scala/leetcode/editor/cn/[724]寻找数组的中心下标.scala
package leetcode.editor.cn
object FindPivotIndex{
//leetcode submit region begin(Prohibit modification and deletion)
object Solution {
    def pivotIndex(nums: Array[Int]): Int =
      val preSums = Array.fill[Int](nums.length+1)(0)
      for (i <- 1 until preSums.length)
        preSums(i) = preSums(i-1) + nums(i-1)

      nums.indices
        .find(i => preSums(i) == preSums.last - preSums(i+1))
        .getOrElse(-1)
}
//leetcode submit region end(Prohibit modification and deletion)

}
